Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/BasicThresholder.java

```diff
@@ -243,11 +243,8 @@ protected double longTermDeviation(TransformMethod method, int shingleSize) {
             return min(sqrt(2.0) * thresholdDeviation.getDeviation(), primaryDeviation.getDeviation());
         } else {
             double first = primaryDeviation.getDeviation();
-            if (method != TransformMethod.NORMALIZE) {
-                first = min(first, sqrt(2.0) * thresholdDeviation.getDeviation());
-
-            }
-            // there is a role of differenceing; either by shingling or by explicit
+            first = min(first, max(secondaryDeviation.getDeviation(), sqrt(2.0) * thresholdDeviation.getDeviation()));
```

Comment:
Yes. The primary deviation actually becomes huge for a level shift -- since the scores are continually large. In this (and even in other scenario) the difference (secondary) can become very small; likewise thresholded may not get sufficiently many samples (at least initially).